### DISCUSSION TITLE
Easy installation vs easy execution

### DISCUSSION THREAD

**Discussion by quintesse** [jbangdev/jbang#259] [General]
TLDR; Try to make Jbang even easier to use by a) making it even easier to install or/and b) making it easier to use without needing installation

So this is something we've been talking about privately quite a bit. The idea is that right now it's pretty easy to install and start using Jbang, especially now that it can automatically install Java for the user. 

But for many users it will still be a 2-step process:

 1. install Sdkman/Scoop/Brew/Choco/etc
 2. use the tool from the first step to install Jbang

It's not hard of course but for many users it might be too much and an obstacle to using Jbang.

So we've been thinking about two options:

 1. A simple `curl get.jbang.dev | sh` experience as used by many tools that would just require you to copy & paste that single line into a CLI to get Jbang installed. (Jbang is a CLI tool so the user needs familiarity with the CLI anyway, so that should not be an obstacle).
 2. A direct execution option, eg `curl run.jbang.dev | sh http://url.to/some/source.java arg1 arg2` that would allow you to execute Jbang and the script in a single line that can be copy & pasted into a CLI.

But what do you think would be the most useful (meaning: where should we focus our energy)? The first option, the second, both, none?

**Comment by maxandersen** [comment]
I would add that that these should in theory also allow to add a bootstrap header to a jbang script (at least on linux/osx) which let as user just download a script and the mere act of running the script (i.e. './hello.java') will result in jbang to be installed and made available - at least for the execution of the script.
